etcdserver/api/membership/cluster.go

```diff
@@ -262,29 +262,38 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 	}
 	switch cc.Type {
 	case raftpb.ConfChangeAddNode, raftpb.ConfChangeAddLearnerNode:
-		if members[id] != nil {
-			return ErrIDExists
-		}
 		urls := make(map[string]bool)
 		for _, m := range members {
 			for _, u := range m.PeerURLs {
 				urls[u] = true
 			}
 		}
-		m := new(Member)
+
+		m := new(ConfigChangeContext)
 		if err := json.Unmarshal(cc.Context, m); err != nil {
 			if c.lg != nil {
 				c.lg.Panic("failed to unmarshal member", zap.Error(err))
 			} else {
 				plog.Panicf("unmarshal member should never fail: %v", err)
 			}
 		}
-		for _, u := range m.PeerURLs {
-			if urls[u] {
-				return ErrPeerURLexists
+		// A ConfChangeAddNode to a existing learner node promotes it to a voting member.
+		if m.IsPromote {
```

Comment:
What if the member is already a voting member?

Reply:
good point

Reply:
will fix that